### PR TITLE
Added SegmentationSlice and new segments

### LINKED ISSUE #82: [Feat]: update segmentation filename generation to allow aliases for segment values
### Feature Type

- [x] Adding new functionality

- [ ] Changing existing functionality

- [ ] Removing existing functionality

### Problem Description

Generating filenames includes the segment name (or alias) and the integer value e.g. "p1", "m3", "ca1" etc. For some segments there are specific aliases for each value which would make sense e.g. for direction currently it is "pa1" but "hb" would be more informative.

### Feature Description

Allowing aliases to be defined for the individual values of a segment used for slice naming. This should be optional and should use aliases for naming if they are present but revert to the old method of segment name and integer value if not present.

The alias for the value should not include the segment name e.g. instead of "m3" it might just be "car" not "mcar".

### Alternative Solutions

Generating the filename using the Segmentation method and then using str replace to update the name.

### Additional Context

Example of the alias in the segment config file.
```yaml
name: direction
alias: pa
values:
  0: non-home-based
  1: home-based
value_alias:
  0: nhb
  1: hb
```

Direction and direction_od are the segments that will benefit from this the most.

### PR DESCRIPTION
**Pull request by MattBuckley-TfN**
# Changes

- Added new `SegmentationSlice` class to hold parameters for a single slice (row) of the segmentation. Class is hashable and provides some simple conversions to tuples.
- Updated relevant `Segmentation` methods to return `SegmentationSlice` instead of `dict[str, int]`
- Added optional segment values aliases e.g. "nhb" instead of "pa0"
- Added new segments
  - userclass
  - direction
  - direction_od

- Closes #82 

# Tasks

- [ ] Have unittests been added (testing individual functions and their edge cases)
- [ ] Has documentation (including user guide) been updated
- [ ] Have new dependencies been added (or changed) in relevant `requirements.txt`
- [ ] Code linting, tests and other workflows pass

### CODE REVIEW

**Review by isaac-tfn — APPROVED** [review]
Looks good, just need to sort out some issues with segments at some point but not in this PR

File: src/caf/base/segments/userclass.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: userclass
```
[the remark this answers is not in the document]
Comment (MattBuckley-TfN) [reply src/caf/base/segments/userclass.yml:1]: Are you happy for this to be addressed in the segment tidy up PR you mention below or shall I change the name now?
